src/Illuminate/Console/GeneratorCommand.php

```diff
@@ -89,11 +89,27 @@ protected function qualifyClass($name)
         $name = str_replace('/', '\\', $name);
 
         return $this->qualifyClass(
-            $this->getDefaultNamespace(trim($rootNamespace, '\\')).'\\'.$name
+            $this->getFinalNamespace(trim($rootNamespace, '\\')).'\\'.$name
         );
     }
 
     /**
+     * Get the final namespace for the class.
+     *
+     * @param  string  $rootNamespace
+     * @return string
+     */
+    protected function getFinalNamespace($rootNamespace)
+    {
+        $userNamespace = config('generators.namespaces.'.str_slug($this->type), null);
```

Comment:
Should be `Str::slug` and the config should be loaded from `$this->laravel` I believe